packages/block-library/src/gallery/editor.scss

```diff
@@ -1,4 +1,7 @@
-:where(figure.wp-block-gallery) {
+// Adding `figure` to the selector increases the specificity above the global
+// styles specificity, which is levelled at 0-1-0. We should figure out why
+// `figure` is needed.
+:root :where(figure.wp-block-gallery) {
```

Comment:
I just added `:root` as a precaution here to make sure this overrides any reset styles.